### ISSUE TITLE
randomly GPU process crash on mac os big sur

### ISSUE THREAD

**Issue by tuxianchao** [electron/electron#27081]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
    
  * 11.0.3
* **Operating System:**
  
  * macOS 11.0.1 / macOS 11.1
* **Last Known Working Electron version:**
  * 
    before update macOS bigsur,it's work fine.
### Expected Behavior

my application rely on WEBGL.it's work fine on macOS version before big sur(11.0.1/11.1). but i upgrade macOS version to big sur. when i use WEBGL,GPU process randomly crash.
### Actual Behavior

 it's should work normal  like on old macOS version.
### To Reproduce

no certain way to recreate this problem，open my electron application(use webgl)，after a while ,randomly will get GPU process crash（ I get webglContextlost in javascript and macos system report gpu proces crash）. 

this crash into four types.

*  Crashed Thread:       3  ThreadPoolForegroundWorker  Dispatch queue: librariesCacheSerialQueue
* Crashed Thread:        0  CrGpuMain  Dispatch queue: com.apple.main-thread
* Crashed Thread:        7  VizCompositorThread
* Crashed Thread:        0
 

### Additional Information

I use electron version 11.0.3 . and cut out some important  infomation.

it's seem like #26687 

try a lot of way to solve it，but no get. Hope for a solution. Thanks.

```
Date/Time:             2020-12-19 22:24:40.566 +0800
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Anonymous UUID:        CF1FA6AE-1FAF-CBCA-ABC1-58B6808551AC

Sleep/Wake UUID:       A700D007-3A47-4775-8BB3-3C1AF4733A99

Time Awake Since Boot: 730000 seconds
Time Since Wake:       24000 seconds

System Integrity Protection: enabled

Crashed Thread:        3  ThreadPoolForegroundWorker  Dispatch queue: librariesCacheSerialQueue

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called
XXXX Helper (GPU)(69519,0x70000831d000) malloc: Incorrect checksum for freed object 0x7fd8d9ff1388: probably modified after being freed.
Corrupt value: 0x2020202273656372

hread 3 Crashed:: ThreadPoolForegroundWorker  Dispatch queue: librariesCacheSerialQueue
0   libsystem_kernel.dylib        	0x00007fff2030e462 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff2033c610 pthread_kill + 263
2   libsystem_c.dylib             	0x00007fff2028f720 abort + 120
3   libsystem_malloc.dylib        	0x00007fff20170430 malloc_vreport + 548
4   libsystem_malloc.dylib        	0x00007fff20184702 malloc_zone_error + 183
5   libsystem_malloc.dylib        	0x00007fff20165502 tiny_malloc_from_free_list + 1805
6   libsystem_malloc.dylib        	0x00007fff20164876 tiny_malloc_should_clear + 233
7   libsystem_malloc.dylib        	0x00007fff20163793 szone_malloc_should_clear + 66
8   com.github.Electron.framework 	0x000000010d5f29b4 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6310660
9   com.github.Electron.framework 	0x000000010d5f29b4 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6310660
10  libsystem_malloc.dylib        	0x00007fff2017ceb7 _malloc_zone_calloc + 59
11  libobjc.A.dylib               	0x00007fff201d8b0b class_createInstance + 72
12  libdispatch.dylib             	0x00007fff20190ef1 _os_object_alloc_realized + 25
13  libsystem_trace.dylib         	0x00007fff200b1259 os_log_create + 367
14  libsystem_coreservices.dylib  	0x00007fff2a2515da ___log_block_invoke + 23
15  libdispatch.dylib             	0x00007fff201917c7 _dispatch_client_callout + 8
16  libdispatch.dylib             	0x00007fff2019296b _dispatch_once_callout + 20
17  libsystem_coreservices.dylib  	0x00007fff2a24fc25 _dirhelper + 2005
18  libsystem_c.dylib             	0x00007fff20211de1 confstr + 457
19  com.apple.Metal               	0x00007fff2831dd3f getCacheFolder(char*, unsigned long, char const*) + 48
20  com.apple.Metal               	0x00007fff2831e004 getCacheMainFolder(char*, unsigned long) + 86
21  com.apple.Metal               	0x00007fff283011d9 MTLCompilerFSCache::openSync() + 125
22  com.apple.Metal               	0x00007fff2831e9ed invocation function for block in MTLCompilerFSCache::getElement(MTLUINT256_t, void**, unsigned long*) + 46
23  libdispatch.dylib             	0x00007fff201917c7 _dispatch_client_callout + 8
24  libdispatch.dylib             	0x00007fff2019e605 _dispatch_lane_barrier_sync_invoke_and_complete + 60
25  com.apple.Metal               	0x00007fff28301093 MTLCompilerFSCache::getElement(MTLUINT256_t, void**, unsigned long*) + 353
26  com.apple.Metal               	0x00007fff2836e734 MultiLevelPipelineCache::getElement(MTLUINT256_t const&) const + 178
27  com.apple.Metal               	0x00007fff2836e014 MTLLibraryBuilder::initLibraryContainerWithRequestData(MTLLibraryContainer*, void*, unsigned long, MTLBuildRequestType, MTLCompileOptions const*, MTLCompilerFunctionRequest const*, unsigned int, bool, void () block_pointer) + 1102
28  com.apple.Metal               	0x00007fff2836f0d4 MTLLibraryBuilder::newLibraryWithRequestData(void*, unsigned long, MTLBuildRequestType, MTLCompileOptions const*, MTLCompilerFunctionRequest const*, unsigned int, bool, void (id<MTLLibrary>, NSError*) block_pointer) + 254
29  com.apple.Metal               	0x00007fff28370bbc MTLLibraryBuilder::newLibraryWithSource(id<MTLDevice>, NSString*, MTLCompileOptions*, bool, void (id<MTLLibrary>, NSError*) block_pointer) + 5952
30  com.apple.Metal               	0x00007fff28348121 -[_MTLDevice newLibraryWithSource:options:completionHandler:] + 75
31  com.github.Electron.framework 	0x0000000111650b25 v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 23129029
32  com.github.Electron.framework 	0x0000000111651700 v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 23132064
33  com.github.Electron.framework 	0x000000010d5adee9 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6029369
34  com.github.Electron.framework 	0x000000010d5c6827 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6130039
35  com.github.Electron.framework 	0x000000010d5c61af v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6128383
36  com.github.Electron.framework 	0x000000010d5e856c v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6268604
37  com.github.Electron.framework 	0x000000010d5c5cf6 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6127174
38  com.github.Electron.framework 	0x000000010d5cd14a v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6156954
39  com.github.Electron.framework 	0x000000010d5ccf0d v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6156381
40  com.github.Electron.framework 	0x000000010d5e8a23 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6269811
41  libsystem_pthread.dylib       	0x00007fff2033c950 _pthread_start + 224
42  libsystem_pthread.dylib       	0x00007fff2033847b thread_start + 15
```

```
Date/Time:             2020-12-19 22:30:41.206 +0800
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Anonymous UUID:        CF1FA6AE-1FAF-CBCA-ABC1-58B6808551AC

Sleep/Wake UUID:       A700D007-3A47-4775-8BB3-3C1AF4733A99

Time Awake Since Boot: 730000 seconds
Time Since Wake:       24000 seconds

System Integrity Protection: enabled

Crashed Thread:        3  ThreadPoolForegroundWorker  Dispatch queue: libraryBuilder

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
dyld: in dlopen_preflight()
abort() called
XXXX Helper (GPU)(69782,0x70000c8cc000) malloc: Incorrect checksum for freed object 0x7ff7ce64b6b0: probably modified after being freed.
Corrupt value: 0x756f7365522f7374



Thread 3 Crashed:: ThreadPoolForegroundWorker  Dispatch queue: libraryBuilder
0   libsystem_kernel.dylib        	0x00007fff2030e462 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff2033c610 pthread_kill + 263
2   libsystem_c.dylib             	0x00007fff2028f720 abort + 120
3   libsystem_malloc.dylib        	0x00007fff20170430 malloc_vreport + 548
4   libsystem_malloc.dylib        	0x00007fff20184702 malloc_zone_error + 183
5   libsystem_malloc.dylib        	0x00007fff20168182 tiny_free_list_add_ptr + 1224
6   libsystem_malloc.dylib        	0x00007fff2016545d tiny_malloc_from_free_list + 1640
7   libsystem_malloc.dylib        	0x00007fff20164876 tiny_malloc_should_clear + 233
8   libsystem_malloc.dylib        	0x00007fff20163793 szone_malloc_should_clear + 66
9   com.github.Electron.framework 	0x000000010c7399b4 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6310660
10  com.github.Electron.framework 	0x000000010c7399b4 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6310660
11  libsystem_malloc.dylib        	0x00007fff2017ceb7 _malloc_zone_calloc + 59
12  libobjc.A.dylib               	0x00007fff201d8b0b class_createInstance + 72
13  libdispatch.dylib             	0x00007fff20190ef1 _os_object_alloc_realized + 25
14  libdispatch.dylib             	0x00007fff20192150 dispatch_group_create + 21
15  com.apple.Metal               	0x00007fff282fb57a invocation function for block in MTLLibraryBuilder::findOrCreateLibraryContainer(MTLUINT256_t const&, bool*) + 238
16  libdispatch.dylib             	0x00007fff201917c7 _dispatch_client_callout + 8
17  libdispatch.dylib             	0x00007fff2019e605 _dispatch_lane_barrier_sync_invoke_and_complete + 60
18  com.apple.Metal               	0x00007fff282fb444 MTLLibraryBuilder::findOrCreateLibraryContainer(MTLUINT256_t const&, bool*) + 148
19  com.apple.Metal               	0x00007fff2836f051 MTLLibraryBuilder::newLibraryWithRequestData(void*, unsigned long, MTLBuildRequestType, MTLCompileOptions const*, MTLCompilerFunctionRequest const*, unsigned int, bool, void (id<MTLLibrary>, NSError*) block_pointer) + 123
20  com.apple.Metal               	0x00007fff28370bbc MTLLibraryBuilder::newLibraryWithSource(id<MTLDevice>, NSString*, MTLCompileOptions*, bool, void (id<MTLLibrary>, NSError*) block_pointer) + 5952
21  com.apple.Metal               	0x00007fff28348121 -[_MTLDevice newLibraryWithSource:options:completionHandler:] + 75
22  com.github.Electron.framework 	0x0000000110797b25 v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 23129029
23  com.github.Electron.framework 	0x0000000110798700 v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 23132064
24  com.github.Electron.framework 	0x000000010c6f4ee9 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6029369
25  com.github.Electron.framework 	0x000000010c70d827 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6130039
26  com.github.Electron.framework 	0x000000010c70d1af v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6128383
27  com.github.Electron.framework 	0x000000010c72f56c v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6268604
28  com.github.Electron.framework 	0x000000010c70ccf6 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6127174
29  com.github.Electron.framework 	0x000000010c71414a v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6156954
30  com.github.Electron.framework 	0x000000010c713f0d v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6156381
31  com.github.Electron.framework 	0x000000010c72fa23 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6269811
32  libsystem_pthread.dylib       	0x00007fff2033c950 _pthread_start + 224
33  libsystem_pthread.dylib       	0x00007fff2033847b thread_start + 15
```

```
Date/Time:             2020-12-19 17:13:11.237 +0800
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Anonymous UUID:        CF1FA6AE-1FAF-CBCA-ABC1-58B6808551AC

Sleep/Wake UUID:       A700D007-3A47-4775-8BB3-3C1AF4733A99

Time Awake Since Boot: 720000 seconds
Time Since Wake:       5300 seconds

System Integrity Protection: enabled

Notes:                 Translocated Process

Crashed Thread:        0  CrGpuMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
abort() called
xxxx Helper (GPU)(55031,0x10df58e00) malloc: Incorrect checksum for freed object 0x7fb6c1ced8c0: probably modified after being freed.
Corrupt value: 0x6372756f7365522f
 

Thread 0 Crashed:: CrGpuMain  Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff2030e462 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff2033c610 pthread_kill + 263
2   libsystem_c.dylib             	0x00007fff2028f720 abort + 120
3   libsystem_malloc.dylib        	0x00007fff20170430 malloc_vreport + 548
4   libsystem_malloc.dylib        	0x00007fff20184702 malloc_zone_error + 183
5   libsystem_malloc.dylib        	0x00007fff20168182 tiny_free_list_add_ptr + 1224
6   libsystem_malloc.dylib        	0x00007fff201677aa tiny_free_no_lock + 1116
7   libsystem_malloc.dylib        	0x00007fff201671f9 free_tiny + 442
8   com.github.Electron.framework 	0x0000000105c8251e v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 18515374
9   com.github.Electron.framework 	0x0000000105c821e6 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 18514550
10  com.github.Electron.framework 	0x0000000105c8199e v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 18512430
11  com.github.Electron.framework 	0x0000000105c80dd2 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 18509410
12  com.github.Electron.framework 	0x0000000109333a87 v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 25654535
13  com.github.Electron.framework 	0x00000001093344ae v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 25657134
14  com.github.Electron.framework 	0x0000000109204dcf v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 24414287
15  com.github.Electron.framework 	0x0000000109214948 v8::internal::compiler::ZoneStats::GetCurrentAllocatedBytes() const + 24478664
16  com.github.Electron.framework 	0x0000000105abcf8f v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 16658463
17  com.github.Electron.framework 	0x0000000106941f4f v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 31883231
18  com.github.Electron.framework 	0x000000010693fd6a v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 31874554
19  com.github.Electron.framework 	0x00000001069438fe v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 31889806
20  com.github.Electron.framework 	0x0000000105ac26c4 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 16680788
21  com.github.Electron.framework 	0x0000000105099fa9 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6029369
22  com.github.Electron.framework 	0x00000001050a7715 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6084517
23  com.github.Electron.framework 	0x00000001050a7320 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6083504
24  com.github.Electron.framework 	0x00000001050d9260 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6288112
25  com.github.Electron.framework 	0x00000001050d5d7a v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6274570
26  com.github.Electron.framework 	0x00000001050d8c3f v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6286543
27  com.apple.CoreFoundation      	0x00007fff204349fc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
28  com.apple.CoreFoundation      	0x00007fff20434964 __CFRunLoopDoSource0 + 180
29  com.apple.CoreFoundation      	0x00007fff204346df __CFRunLoopDoSources0 + 248
30  com.apple.CoreFoundation      	0x00007fff20433111 __CFRunLoopRun + 890
31  com.apple.CoreFoundation      	0x00007fff204326be CFRunLoopRunSpecific + 563
32  com.apple.Foundation          	0x00007fff211bcfa1 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212
33  com.github.Electron.framework 	0x00000001050d9729 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6289337
34  com.github.Electron.framework 	0x00000001050d87d2 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6285410
35  com.github.Electron.framework 	0x00000001050a7c81 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6085905
36  com.github.Electron.framework 	0x0000000105086379 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5948425
37  com.github.Electron.framework 	0x0000000106a1007a v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 32727306
38  com.github.Electron.framework 	0x0000000104cc0c7d v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 1993997
39  com.github.Electron.framework 	0x00000001045042be ElectronInitializeICUandStartNode + 6186878
40  com.github.Electron.framework 	0x00000001045047c2 ElectronInitializeICUandStartNode + 6188162
41  com.github.Electron.framework 	0x0000000103f1db08 ElectronMain + 136
42  com.kakacode-student.app.helper.GPU	0x0000000103eb4d40 0x103eb1000 + 15680
43  libdyld.dylib                 	0x00007fff20357631 start + 1
```

```
Date/Time:             2020-12-19 19:11:00.812 +0800
OS Version:            macOS 11.0.1 (20B29)
Report Version:        12
Anonymous UUID:        CF1FA6AE-1FAF-CBCA-ABC1-58B6808551AC

Sleep/Wake UUID:       A700D007-3A47-4775-8BB3-3C1AF4733A99

Time Awake Since Boot: 720000 seconds
Time Since Wake:       12000 seconds

System Integrity Protection: enabled

Crashed Thread:        7  VizCompositorThread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       EXC_I386_GPFLT
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [61113]


Thread 7 Crashed:: VizCompositorThread
0   libsystem_malloc.dylib        	0x00007fff20167e45 tiny_free_list_add_ptr + 395
1   libsystem_malloc.dylib        	0x00007fff201677aa tiny_free_no_lock + 1116
2   libsystem_malloc.dylib        	0x00007fff201671f9 free_tiny + 442
3   com.github.Electron.framework 	0x000000010ee273c7 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 14681175
4   com.github.Electron.framework 	0x000000010ef34bd9 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 15785065
5   com.github.Electron.framework 	0x000000010ef349d6 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 15784550
6   com.github.Electron.framework 	0x000000010ef342e5 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 15782773
7   com.github.Electron.framework 	0x000000010ef33986 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 15780374
8   com.github.Electron.framework 	0x000000010ff4ba59 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 32656105
9   com.github.Electron.framework 	0x000000010ff4aea5 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 32653109
10  com.github.Electron.framework 	0x000000010ff39e0b v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 32583323
11  com.github.Electron.framework 	0x000000010ff374e7 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 32572791
12  com.github.Electron.framework 	0x000000010ff37418 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 32572584
13  com.github.Electron.framework 	0x000000010d9a8f84 ElectronInitializeICUandStartNode + 5497924
14  com.github.Electron.framework 	0x000000010e76c6ea v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 7624570
15  com.github.Electron.framework 	0x000000010e7721be v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 7647822
16  com.github.Electron.framework 	0x000000010e771aef v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 7646079
17  com.github.Electron.framework 	0x000000010e76a2ad v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 7615293
18  com.github.Electron.framework 	0x000000010e76a99e v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 7617070
19  com.github.Electron.framework 	0x000000010e7806b9 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 7706441
20  com.github.Electron.framework 	0x000000010e5e6fa9 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6029369
21  com.github.Electron.framework 	0x000000010e5f4715 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6084517
22  com.github.Electron.framework 	0x000000010e5f4320 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6083504
23  com.github.Electron.framework 	0x000000010e5bccc5 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5856597
24  com.github.Electron.framework 	0x000000010e5f4c81 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6085905
25  com.github.Electron.framework 	0x000000010e5d3379 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 5948425
26  com.github.Electron.framework 	0x000000010e60ccbf v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6184271
27  com.github.Electron.framework 	0x000000010e621ae3 v8::internal::SetupIsolateDelegate::SetupHeap(v8::internal::Heap*) + 6269811
28  libsystem_pthread.dylib       	0x00007fff2033c950 _pthread_start + 224
29  libsystem_pthread.dylib       	0x00007fff2033847b thread_start + 15
```

**Comment by tuxianchao** [comment]
I'm a web developer .no rich  knowledge and experience of c++. 

Hope for a solution. Thanks

**Comment by skylingfly** [comment]
+1, since we updated this new system: mac os big sur,  got few crash also.

**Comment by XuJinyuan** [comment]
+1，Electron will crash randomly when I was using Intel GPU. AMD GPU was OK (MBP 15 2018)

**Comment by wenssh** [comment]
+1

**Comment by wenssh** [comment]
@codebytere @skylingfly @tuxianchao  this issue is a bug in chrome, they fix it in https://chromium-review.googlesource.com/c/chromium/src/+/2526161

**Comment by tuxianchao** [comment]
> @codebytere @skylingfly @tuxianchao this issue is a bug in chrome, they fix it in https://chromium-review.googlesource.com/c/chromium/src/+/2526161

We checked this commit in our src of electron, which the chromium src is "HEAD detached from 87.0.4200.67".
The modifications mentioned by "2526161" are already in 87.0.4200.67.
So the GPU crash is caused by other reason not yet discovered
The same program running on Chromium of version "87.0.4200.67" will not crash at all, 
even if the chromium is build directly from src at tag "87.0.4200.67".
So maybe the crash is not related to chromium.

**Comment by luckman212** [comment]
How about on Electron 12.x or 13.x? Anyone do any testing?

**Comment by nornagon** [comment]
The commit linked in https://github.com/electron/electron/issues/27081#issuecomment-766803756 landed [in Chrome 87.0.4280.60](https://storage.googleapis.com/chromium-find-releases-static/996.html#9969257bc42b4b621e26a4b869f942324c2c2e0e), which was included in Electron with https://github.com/electron/electron/commit/e4f3f9e1dbed451f438e29ef925f11ad5109ba79, which was in the first v11.0.0 release. So indeed this seems to be different to the chromium issue linked, though the version numbers @tuxianchao mentioned are a little off.

**Comment by nornagon** [comment]
Duplicate of #26686
